util/src/main/java/io/camunda/zeebe/util/FatalErrorHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import org.slf4j.Logger;
+
+public final class FatalErrorHandler implements UncaughtExceptionHandler {
+  private final Logger log;
+
+  public FatalErrorHandler(final Logger log) {
+    this.log = log;
+  }
+
+  public void handleError(final Throwable e) {
```

Comment:
:x: I normally wouldn't push for changing method names, but in this case since we do exit (which is quite extreme), I would keep it in the name so it's very clear what happens when you use it :smile: wdyt?

Reply:
I agree that we should document this but I slightly disagree that we have to reflect that in the method name. As a user of `FatalErrorHandler`, I'd like to express: "Okay, some error happened, please make a decision if anything should be done about this" That's just my perspective and the reason why I kept the method name so generic. I can understand your point though, so your decision.

Reply:
That's fair, I just think exiting is a pretty extreme side effect and typically unexpected, so I would save the surprise by simply naming it `exitOnFatalError` or `exitIfFatalError` or some such, so a reader wouldn't even have to look into the implementation (except maybe to know what is or isn't a fatal error). But I'm happy to be proven wrong and maybe it's not needed, we can go with this and modify it if it does cause confusion :slightly_smiling_face: